src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,109 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_QUERY =
+            "INSERT INTO books (id, title, price) VALUES (?, ?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_QUERY = "DELETE FROM books WHERE id = ?";
+    private static final String URL = "jdbc:mysql://localhost:3306/book";
+    private static final String LOGIN = "root";
+    private static final String PASSWORD = "root";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = DriverManager.getConnection(URL, LOGIN, PASSWORD)) {
```

Comment:
It is better to create a connection to the database in a separate util class. Same in the other methods.